src/main/java/net/pms/dlna/protocolinfo/AribOrJpProfileName.java

```diff
@@ -0,0 +1,147 @@
+ */
+@SuppressWarnings("checkstyle:InterfaceIsType")
+public interface AribOrJpProfileName extends ProfileName {
+
+	/** The static {@code NONE} instance representing a blank/empty value */
+	AribOrJpProfileName NONE = new DefaultAribOrJpProfileName("");
+
+	/**
+	 * The static factory singleton instance used to create and retrieve
+	 * {@link AribOrJpProfileName} instances.
+	 */
+	AribOrJpProfileNameFactory FACTORY = new AribOrJpProfileNameFactory();
+
+	/** The static attribute name always used for this class */
+	ProtocolInfoAttributeName NAME = KnownProtocolInfoAttributeName.ARIB_OR_JP_PN;
+
+	/**
+	 * A factory for creating, caching and retrieving
+	 * {@link AribOrJpProfileName} instances.
+	 */
+	public static class AribOrJpProfileNameFactory extends AbstractProfileNameFactory<AribOrJpProfileName> {
+
+		/**
+		 * Only for internal use only, use {@link AribOrJpProfileName#FACTORY}
+		 * to get the singleton instance.
+		 */
+		protected AribOrJpProfileNameFactory() {
+		}
+
+		@Override
+		protected AribOrJpProfileName getNoneInstance() {
+			return NONE;
+		}
+
+		@Override
+		protected AribOrJpProfileName searchKnownInstances(String value) {
+			// Check for known instances
+			for (KnownAribOrJpProfileName knownAttribute : KnownAribOrJpProfileName.values()) {
+				if (value.equals(knownAttribute.getValue())) {
+					return knownAttribute;
+				}
+			}
+			return null;
+		}
+
+		@Override
+		protected AribOrJpProfileName getNewInstance(String value) {
+			return new DefaultAribOrJpProfileName(value);
+		}
+	}
+
+	/**
+	 * This contains predefined {@code ARIB.OR.JP_PN} values.
+	 */
+	public enum KnownAribOrJpProfileName implements AribOrJpProfileName {
+
+		/**
+		 * MPEG2-Video ({@code ISO/IEC 13818-2}) and MPEG2-AAC (
+		 * {@code ISO/IEC 13818-7}) partial transport stream format with time
+		 * stamp with mime-type
+		 * {@code application/X-arib-cp;CONTENTFORMAT=<content-mimetype>}.
+		 * <p>
+		 * For full specification, see <a href=
+		 * "http://www.arib.or.jp/english/html/overview/doc/6-STD-B21v4_6-E1.pdf"
+		 * >ARIB STD – B21</a> chapter <b>9.2.4</b>.
+		 */
+		MPEG_TTS_CP;
+
+		@Override
+		public String getValue() {
+			return super.toString();
+		}
+
+		@Override
+		public String toString() {
+			return NAME + " = " + super.toString();
+		}
+
+		@Override
+		public ProtocolInfoAttributeName getName() {
+			return NAME;
+		}
+
+		@Override
+		public String getNameString() {
+			return NAME.getName();
+		}
+
+		@Override
+		public String getAttributeString() {
+			return NAME + "=" + super.toString();
+		}
+
+	}
+
+	/**
+	 * This is the default, immutable class implementing
+	 * {@link AribOrJpProfileName}. {@link AribOrJpProfileNameFactory} creates
+	 * and caches instances of this class.
+	 */
+	public static class DefaultAribOrJpProfileName extends AbstractDefaultProfileName implements AribOrJpProfileName {
+
+		private static final long serialVersionUID = 1L;
+
+		/**
+		 * Only for internal use only, use
```

Comment:
@Sami32 You do realize that I've added more than 9000 lines 👅  The Javadocs are extremely boring and repetetive..